tools/generate_bech32_tests.py

```
@@ -0,0 +1,224 @@
+            conv = convertbits(data_bytes, 8, 5, pad=pad)
+        
+        if conv is None:
+            return None
+        
+        # Choose encoding constant
+        spec = BECH32_CONST if encoding == "bech32" else BECH32M_CONST
+        
+        # Encode
+        result = bech32_encode(hrp, conv, spec)
+        return result
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)